pkg/utils/image/infos.go

```diff
@@ -0,0 +1,73 @@
+package image
+
+import (
+	"strings"
+
+	"github.com/distribution/distribution/reference"
+	"github.com/pkg/errors"
+)
+
+type ImageInfo struct {
+	// Registry is the URL address of the image registry e.g. `docker.io`
+	Registry string `json:"registry,omitempty"`
+
+	// Name is the image name portion e.g. `busybox`
+	Name string `json:"name"`
+
+	// Path is the repository path and image name e.g. `some-repository/busybox`
+	Path string `json:"path"`
+
+	// Tag is the image tag e.g. `v2`
+	Tag string `json:"tag,omitempty"`
+
+	// Digest is the image digest portion e.g. `sha256:128c6e3534b842a2eec139999b8ce8aa9a2af9907e2b9269550809d18cd832a3`
+	Digest string `json:"digest,omitempty"`
+}
+
+func (i *ImageInfo) String() string {
+	image := i.Registry + "/" + i.Path
+	if i.Digest != "" {
+		return image + "@" + i.Digest
+	} else {
+		return image + ":" + i.Tag
+	}
+}
+
+func GetImageInfo(image string) (*ImageInfo, error) {
+	image = addDefaultDomain(image)
+	ref, err := reference.Parse(image)
+	if err != nil {
+		return nil, errors.Wrapf(err, "bad image: %s", image)
+	}
+	var registry, path, name, tag, digest string
+	if named, ok := ref.(reference.Named); ok {
+		registry = reference.Domain(named)
+		path = reference.Path(named)
+		name = path[strings.LastIndex(path, "/")+1:]
+	}
+	if tagged, ok := ref.(reference.Tagged); ok {
+		tag = tagged.Tag()
+	}
+	if digested, ok := ref.(reference.Digested); ok {
+		digest = digested.Digest().String()
+	}
+	// set default tag - the domain is set via addDefaultDomain before parsing
+	if digest == "" && tag == "" {
+		tag = "latest"
+	}
+	return &ImageInfo{
+		Registry: registry,
+		Name:     name,
+		Path:     path,
+		Tag:      tag,
+		Digest:   digest,
+	}, nil
```

Comment:
Could we instead use ggcr to parse this reference? That does a much better job of normalizing all of these things and it is also what we use for cosign/imageRegistry integration. https://github.com/google/go-containerregistry/blob/a0c4bd256482b8522065d5f6cf966281ef270680/pkg/name/ref.go#L40

We just need to do 

```go
package main

import (
	"fmt"

	"github.com/google/go-containerregistry/pkg/name"
)

func printRef(refS string) {
	ref, _ := name.ParseReference(refS)
	fmt.Println(ref.Context().RegistryStr())
	fmt.Println(ref.Context().RepositoryStr())
	fmt.Println(ref.Context().Name())

	if tag, ok := ref.(name.Tag); ok {
		fmt.Println(tag.TagStr())
	}

	if digest, ok := ref.(name.Digest); ok {
		fmt.Println(digest.DigestStr())
	}
}

func main() {
	printRef("nginx")
	printRef("busybox@sha256:59603d4d14e08e4a60643588af127f3c3dddf8417bf6b575dd93a3cbe3e48593")
}
```

```
index.docker.io
library/nginx
index.docker.io/library/nginx
latest
index.docker.io
library/busybox
index.docker.io/library/busybox
sha256:59603d4d14e08e4a60643588af127f3c3dddf8417bf6b575dd93a3cbe3e48593
```

The current parsing logic is slight incorrect in terms of conventions and also doesn't account for `library` namespace on dockerhub.